tor/tor.go

```diff
@@ -0,0 +1,166 @@
+package tor
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/miekg/dns"
+	"github.com/roasbeef/btcd/connmgr"
+	"golang.org/x/net/proxy"
+)
+
+const (
+	localhost = "127.0.0.1"
```

Comment:
Fixed this so that callers can specify the host:port of the SOCKS proxy, rather than just the port. Within `lnd`, the default is still set to IPv4 localhost.